go/jwt-go/security/jwt.yaml

```diff
@@ -1,19 +1,32 @@
 rules:
   - id: hardcoded-jwt-key
+    message: >-
+      A hard-coded credential was detected. It is not recommended 
+      to store credentials in source-code, as this risks secrets
+      being leaked and used by either an internal or external malicious adversary. 
+      It is recommended to use environment variables to 
+      securely provide credentials or retrieve credentials from 
+      a secure vault or HSM (Hardware Security Module).
     metadata:
       cwe: "CWE-798: Use of Hard-coded Credentials"
-      owasp: "A2: Broken Authentication"
+      references: 
+        - https://cheatsheetseries.owasp.org/cheatsheets/Secrets_Management_CheatSheet.html
+      owasp:
+        - A07:2021 - Identification and Authentication Failures
+        - A03:2017 - Sensitive Data Exposure
       category: security
       technology:
         - jwt
+        - secrets
       confidence: MEDIUM
-    pattern-either:
-      - pattern: |
-          $X = []byte("...")
-          ...
-          $Y := $TOKEN.SignedString($X)
-      - pattern: |
-          $TOKEN.SignedString([]byte("..."))
-    message: JWT token is hardcoded
-    languages: [go]
     severity: WARNING
+    patterns:
+      - pattern-either:
+        - pattern: |
+            $X = []byte("$F")
```

Comment:
Are we keeping this search mode to handle interprocedural behavior in OSS Semgrep?

Reply:
i think i was lazy let me see if i can convert it into taint

Reply:
updated to taint